var/spack/repos/builtin/packages/planck-likelihood/package.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+import shutil
+
+
+class PlanckLikelihood(Package):
+    """2015 Cosmic Microwave Background (CMB) spectra and likelihood code"""
+
+    homepage = "https://wiki.cosmos.esa.int/planckpla2015/index.php/CMB_spectrum_%26_Likelihood_Code"
+    url      = "http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Code-v2.0.R2.00.tar.bz2"
+
+    version('2.00', '7a081679ff249dc4f94fb7177e16e818')
+
+    patch('fortran.patch')
+    patch('make.patch')
+
+    resource(
+        name='baseline',
+        url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-baseline_R2.00.tar.gz",
+        md5='7e784819cea65dbc290ea3619420295a',
+        destination='.')
+    # resource(
+    #     name='lensing-ext',
+    #     url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-lensing-ext.R2.00.tar.gz",
+    #     md5='091736f73b47a09162050bee27d68399',
+    #     destination='.')
+    # resource(
+    #     name='plik-DS',
+    #     url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-plik-DS.R2.00.tar.gz",
+    #     md5='76ac04f989025eecab3825aba7e41f36',
+    #     destination='.')
+    # resource(
+    #     name='plik-HM-ext',
+    #     url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-plik-HM-ext.R2.00.tar.gz",
+    #     md5='1c3bd8221f973b7bf7e76647451fd6e5',
+    #     destination='.')
+    # resource(
+    #     name='plik-unbinned',
+    #     url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-plik-unbinned.R2.00.tar.gz",
+    #     md5='c5869aa6b6581b6863d2a6e0ffd3826c',
+    #     destination='.')
```

Comment:
The PLC dataset comes in several tarballs. The ones commented out are ones that are likely not needed by most people. I'd like to e.g. add variants to enable these, since some of these tarballs are hugs.

I think I recall that LLVM uses a similar approach. I'll look there for inspiration.